File: perf/zmq-throughput-b.py
import optparse
import zmq

parser = optparse.OptionParser()
parser.add_option("--address", default="tcp://127.0.0.1:5555", help="Local address.  Default: %default")
parser.add_option("--size", type="int", default=1, help="Message size.  Default: %default")
parser.add_option("--count", type="int", default=100000, help="Message count.  Default: %default")
parser.add_option("--hwm", type="int", default=1000000, help="High water mark.  Default: %default")
options, arguments = parser.parse_args()

context = zmq.Context()
output = context.socket(zmq.PUSH)
output.setsockopt(zmq.HWM, options.hwm)
output.connect(options.address)

message = "*" * options.size
for i in range(options.count):
  output.send(message)

File: perf/zmq-throughput-b.cpp
#include <stdexcept>
#include <sstream>
#include <zmq.hpp>

int main(int argc, char* argv[])
{
  if(argc != 5)
    throw std::runtime_error("Usage: <address> <size> <count> <hwm>");
  const std::string address = argv[1];
  const int size = ::atoi(argv[2]);
  const int count = ::atoi(argv[3]);

  uint64_t hwm = 0;
  std::stringstream buffer(argv[4]);
  buffer >> hwm;

  zmq::context_t context(1);
  zmq::socket_t output(context, ZMQ_PUSH);
  output.setsockopt(ZMQ_HWM, &hwm, sizeof(hwm));
  output.connect(address.c_str());

  for(int i = 0; i != count; ++i)
  {
    zmq::message_t message(size);
    output.send(message, 0);
  }

  return 0;
}
